jgrapht-core/src/main/java/org/jgrapht/alg/lca/TarjanLCAFinder.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.lca;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.alg.interfaces.LCAAlgorithm;
+import org.jgrapht.alg.util.Pair;
+import org.jgrapht.alg.util.UnionFind;
+
+import java.util.*;
+
+/**
+ * Tarjan's offline algorithm for computing lowest common ancestors in rooted trees and forests.
+ *
+ * <p>
+ * See the article on
+ * <a href="https://en.wikipedia.org/wiki/Tarjan%27s_off-line_lowest_common_ancestors_algorithm">wikipedia</a> for more
```

Comment:
Aside form a wiki url, do you also have a formal reference? I'm certain JGrapht will exist long after Wikipedia goes extinct :)

Reply:
This comment was marked as resolved but I dont see any changes?

Reply:
There are. See https://github.com/jgrapht/jgrapht/blob/7ca0aeb3629be320b508c2a6eef2b44a790d4b81/jgrapht-core/src/main/java/org/jgrapht/alg/lca/TarjanLCAFinder.java#L39

Am I the only one that sees them? :)